unix/w0vncserver/wlr/WlrVirtualKeyboard.cxx

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2025 Adam Halim for Cendio AB
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "virtual-keyboard-unstable-v1.h"
+
+#include "../w0vncserver.h"
+#include "../wayland/WDisplay.h"
+#include "../wayland/WSeat.h"
+#include "../wayland/WKeyboard.h"
+#include "WlrVirtualKeyboard.h"
+
+WlrVirtualKeyboard::WlrVirtualKeyboard(WDisplay* display, WSeat* seat)
+  : WObject(display, "zwp_virtual_keyboard_manager_v1",
+            &zwp_virtual_keyboard_manager_v1_interface)
+{
+  if (!seat->getKeyboard()->getSize()) {
+    fatal_error("Keyboard keymap is not set");
+    return;
+  }
+
+  manager = (zwp_virtual_keyboard_manager_v1*) boundObject;
+
+  keyboard = zwp_virtual_keyboard_manager_v1_create_virtual_keyboard(manager,
+                                                                     seat->getSeat());
+  if (!keyboard) {
+    fatal_error("Failed to create virtual keyboard");
+    return;
+  }
+
+  zwp_virtual_keyboard_v1_keymap(keyboard,
+                                 seat->getKeyboard()->getFormat(),
+                                 seat->getKeyboard()->getFd(),
+                                 seat->getKeyboard()->getSize());
+}
+
+WlrVirtualKeyboard::~WlrVirtualKeyboard()
+{
+  if (manager)
+    zwp_virtual_keyboard_manager_v1_destroy(manager);
+  if (keyboard)
+    zwp_virtual_keyboard_v1_destroy(keyboard);
+}
+
+void WlrVirtualKeyboard::keyEvent(uint32_t keysym, uint32_t keycode, bool down)
+{
+  zwp_virtual_keyboard_v1_key(keyboard, keysym, keycode, down);
```

Comment:
This seems wrong. Where are things translated from RFB keycode to Wayland keycode?